src/PokemonGymOwner.java

```diff
@@ -0,0 +1,22 @@
+public class PokemonGymOwner extends PokemonGymImpl{
```

Comment:
Je hebt hier extends PokemonGymImpl, maar een PokemonGymOwner is natuurlijk een persoon en geen gym.  Je had hier wel PokemonTrainer kunnen extenden.  In dat geval had je daar alle pokemons kunnen declareren, die had je dan hier geërft.